src/core/unit.c

```diff
@@ -1740,15 +1740,17 @@ int unit_reload(Unit *u) {
 
         unit_add_to_dbus_queue(u);
 
+        if (!UNIT_VTABLE(u)->reload) {
+                unit_notify(u, unit_active_state(u), unit_active_state(u), true);
+                return 0;
+        }
+
         return UNIT_VTABLE(u)->reload(u);
 }
 
 bool unit_can_reload(Unit *u) {
         assert(u);
 
-        if (!UNIT_VTABLE(u)->reload)
-                return false;
-
```

Comment:
hmm, not sure i like this part. I think unit_can_reload() should return true only for units that either natively support reloading (as before), or which have a reload propagation dep set. For plain units that can't reload nor have such a dep we should return false, already to keep difference in behaviour minimal

Reply:
In fact if a unit has such a dep you really should return true here, unconditionally

Reply:
That approch means that reloading a unit could create an error depending on another unit having a ReloadPropagatedFrom.

is that what we want ? It means that you can't just reload a unit, you have to test first if reloading it makes sense....

or should that detail be checked at a higher level than unit_can_reload ?

something like
unit_can_reload : will reloading the unit actually do something
reload_unit : always succeed unless there is a "real" failure when reloading

that's a change of semantic, i'll have to check all the call sites, but I can do that...